devdoc/buffer_requirements.md

```diff
@@ -39,7 +39,7 @@ extern BUFFER_HANDLE BUFFER_create(const unsigned char* source, size_t size);
 
 BUFFER_create creates a new buffer from the memory at source, having size "size".
 **SRS_BUFFER_02_001: [**If source is NULL then BUFFER_create shall return NULL.**]** 
-**SRS_BUFFER_02_002: [**Otherwise, BUFFER_create shall allocate memory to hold size bytes and shall copy from source size bytes into the newly allocated memory.**]** 
+**SRS_BUFFER_02_002: [**Otherwise, BUFFER_create shall allocate memory to hold size bytes and shall copy from source size bytes into the newly allocated memory. If size parameter is 0 then buffer field of BUFFER_TAG shall still hold a valid pointer whose content is '\0' and size field shall be 0.**]** 
```

Comment:
>If size parameter is 0 then buffer field of BUFFER_TAG shall still hold a valid pointer whose content is '\0' and size field shall be 0. [](start = 161, length = 136)

I'd phrase this like:

"If size parameter is 0 then no memory shall be allocated and size for the buffer shall be set to 0."

That's because I think there is no reason to allocate anything, simply setting the internal buffer to NULL should be good.